mesonbuild/scripts/run_tool.py

```diff
@@ -44,29 +112,18 @@ def run_clang_tool(name: str, srcdir: Path, builddir: Path, fn: T.Callable[...,
     suffixes = set(lang_suffixes['c']).union(set(lang_suffixes['cpp']))
     suffixes.add('h')
     suffixes = {f'.{s}' for s in suffixes}
-    futures = []
-    returncode = 0
-    e = concurrent.futures.ThreadPoolExecutor()
-    try:
-        for f in itertools.chain(*globs):
+    for g in globs:
+        for f in g:
```

Comment:
Why remove the `itertoos.chain`? Although this would be nicer as `itertools.chain.from(globs)`